onnx/reference/ops/op_reduce_l1.py

```diff
@@ -3,11 +3,48 @@
 
 import numpy as np
 
+from onnx.defs import onnx_opset_version
+
 from ._op import OpRunReduceNumpy
 
 
-class ReduceL1(OpRunReduceNumpy):
+class ReduceL1_1(OpRunReduceNumpy):
     def _run(self, data, axes=None, keepdims=None):  # type: ignore
         return (
             np.sum(np.abs(data), axis=axes, keepdims=keepdims).astype(dtype=data.dtype),
         )
+
+
+class ReduceL1_11(ReduceL1_1):
+    pass
+
+
+class ReduceL1_13(ReduceL1_1):
+    pass
+
+
+class ReduceL1_18(OpRunReduceNumpy):
+    def run(self, data, axes=None, keepdims=None, noop_with_empty_axes=None):  # type: ignore
+        keepdims = keepdims or self.keepdims  # type: ignore
+        noop_with_empty_axes = noop_with_empty_axes or self.noop_with_empty_axes  # type: ignore
+        return self._run(data, axes, keepdims, noop_with_empty_axes)
+
+    def _run(self, data, axes, keepdims=1, noop_with_empty_axes=0):  # type: ignore
```

Comment:
You may remove `def run` and define the default values for `def _run` as None. If None, the main class automatically looks into the node definition to replace them or the schema if they are not specified.

Reply:
But is this wrong? I find this current form useful since a reader can understand what happens (what is the default value) without having to look elsewhere.

Reply:
@xadupre , I agree with @gramalingam  here. We need at least on call (_run here) to follow precisely the op schema signature.

Reply:
In addition, I think the suggested _run signature does not work with nest calls - when the op is part of other function ops. There for now I like to keep the current code.